app/forms/security_event_form.rb

```diff
@@ -0,0 +1,78 @@
+# Parses RISC Security Event Token (SETs)
+class SecurityEventForm
+  attr_reader :body
+
+  include ActiveModel::Model
+
+  validates_presence_of :payload
+  validates_presence_of :event_type
+  validates_presence_of :user
+
+  # @param [String] body
+  def initialize(body:)
+    @body = body
+  end
+
+  # @return [Array(Boolean, ActiveModel::Errors)]
+  def submit
+    if valid?
+      create_security_event!
+      [true, nil]
+    else
+      [false, errors]
+    end
+  end
+
+  def create_security_event!
+    SecurityEvent.create!(
+      user: user,
+      uuid: payload['jti'],
+      event_type: event_type,
+      issued_at: payload['iat'] ? Time.zone.at(payload['iat']) : nil,
+      raw_event: payload.to_json
+    )
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def payload
+    @payload ||= begin
+      payload = nil
+
+      IdpPublicKeys.all.each do |public_key|
+        payload, _headers = JWT.decode(
+          body, public_key, true, algorithm: 'RS256', leeway: Float::INFINITY
+        )
+        break if payload
+      rescue JWT::DecodeError
+        next
+      end
+
+      return payload if payload
+
+      errors.add(:jwt, 'could not verify JWT with any known keys')
+      {}
+    end
+  end
+  # rubocop:enable Metrics/MethodLength
```

Comment:
I vote to remove or increase this limit

Reply:
This codebase still uses CodeClimate to lint... and the `.rubocop.yml` is so far out of date I can't even run rubocop locally... unsure where to change that limit 😬

Reply:
I will take a look into this